test/unit/app/controllers/incoming-transactions-test.js

```diff
@@ -139,28 +182,443 @@ describe('IncomingTransactionsController', function () {
+        blockTracker: getMockBlockTracker(),
+        networkController: getMockNetworkController(ROPSTEN),
+        preferencesController: getMockPreferencesController(),
+        initState: getNonEmptyInitState(),
+      })
+      // reply with a valid request for any supported network, so that this test has every opportunity to fail
+      for (const network of [GOERLI, KOVAN, MAINNET, RINKEBY, ROPSTEN]) {
+        nock(`https://api${network === MAINNET ? '' : `-${network.toLowerCase()}`}.etherscan.io`)
+          .get(/api.+/u)
+          .reply(
+            200,
+            JSON.stringify({
+              status: '1',
+              result: [getFakeEtherscanTransaction()],
+            }),
+          )
+      }
+      const updateStateStub = sinon.stub(incomingTransactionsController.store, 'updateState')
+      const updateStateCalled = waitUntilCalled(updateStateStub, incomingTransactionsController.store)
+      const putStateStub = sinon.stub(incomingTransactionsController.store, 'putState')
+      const putStateCalled = waitUntilCalled(putStateStub, incomingTransactionsController.store)
+
+      incomingTransactionsController.stop()
+
+      try {
+        await Promise.race([
+          updateStateCalled,
+          putStateCalled,
+          new Promise((_, reject) => {
+            setTimeout(() => reject(new Error('TIMEOUT')), 1000)
+          }),
+        ])
+        assert.fail('Update state should not have been called')
+      } catch (error) {
+        assert(error.message === 'TIMEOUT', 'setState was not called')
+      }
+    })
+
+    it('should update when the selected address changes and on supported network', async function () {
+      const incomingTransactionsController = new IncomingTransactionsController({
+        blockTracker: getMockBlockTracker(),
+        networkController: getMockNetworkController(ROPSTEN),
+        preferencesController: getMockPreferencesController(),
+        initState: getNonEmptyInitState(),
+      })
+      const NEW_MOCK_SELECTED_ADDRESS = `${MOCK_SELECTED_ADDRESS}9`
+      const startBlock = getNonEmptyInitState().incomingTxLastFetchedBlocksByNetwork[ROPSTEN]
+      nock('https://api-ropsten.etherscan.io')
+        .get(`/api?module=account&action=txlist&address=${NEW_MOCK_SELECTED_ADDRESS}&tag=latest&page=1&startBlock=${startBlock}`)
+        .reply(
+          200,
+          JSON.stringify({
+            status: '1',
+            result: [getFakeEtherscanTransaction(NEW_MOCK_SELECTED_ADDRESS)],
+          }),
+        )
+      const updateStateStub = sinon.stub(incomingTransactionsController.store, 'updateState')
+      const updateStateCalled = waitUntilCalled(updateStateStub, incomingTransactionsController.store)
+
+      const subscription = incomingTransactionsController.preferencesController.store.subscribe.getCall(1).args[0]
+      await subscription({ selectedAddress: MOCK_SELECTED_ADDRESS }) // first update is always skipped
```

Comment:
The meaning of this comment is a little unclear. Skipped how? An uninformed reader (👋) can surmise that `updateState` won't be called after the first update, but is that it?

Reply:
That is correct - `updateState` won't be called after the first update.

I don't think this behavior was intended - I was planning to treat it as a bug and fix it in a later PR. But yeah, that's where we're at right now.

Reply:
I can expand on the comment for now, in case fixing this properly ends up getting delayed.

Reply:
Addressed here: https://github.com/MetaMask/metamask-extension/pull/9755/commits/28a3d43098e8ae22a8e6176f7ba17f9bda0887e3